File: frontend/src/types/tamVang.ts
// src/types/tamVang.ts

import { z } from 'zod';

export const tamVangSchema = z.object({
  nhanKhauId: z.number(),
  ngayBatDau: z.string().nonempty(),
  ngayKetThuc: z.string().nonempty(),
  noiDen: z.string().optional(),
  lyDo: z.string().optional(),
});

export type TamVang = z.infer<typeof tamVangSchema> & { id: number };
export type TamVangFormValues = z.infer<typeof tamVangSchema>;

File: backend/api/src/main/java/com/quanlynhankhau/api/entity/TamVang.java
// src/main/java/com/quanlynhankhau/api/entity/TamVang.java

package com.quanlynhankhau.api.entity;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "tam_vang")
@Getter
@Setter
public class TamVang {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "nhan_khau_id")
    private NhanKhau nhanKhau;

    @Column(name = "ngay_bat_dau", nullable = false)
    private LocalDate ngayBatDau;

    @Column(name = "ngay_ket_thuc", nullable = false)
    private LocalDate ngayKetThuc;

    @Column(name = "noi_den")
    private String noiDen;

    @Column(name = "ly_do", columnDefinition = "TEXT")
    private String lyDo;

    @Column(name = "ngay_cap")
    private LocalDate ngayCap;

    @Column(name = "nguoi_cap")
    private String nguoiCap;
}
